tests/integration/security/authorization_test.go

```diff
@@ -847,123 +864,110 @@ func TestAuthorization_Conditions(t *testing.T) {
 	framework.NewTest(t).
 		Features("security.authorization.conditions").
 		Run(func(ctx framework.TestContext) {
-			nsA := namespace.NewOrFail(t, ctx, namespace.Config{
-				Prefix: "v1beta1-conditions-a",
-				Inject: true,
-			})
-			nsB := namespace.NewOrFail(t, ctx, namespace.Config{
-				Prefix: "v1beta1-conditions-b",
-				Inject: true,
-			})
-			nsC := namespace.NewOrFail(t, ctx, namespace.Config{
-				Prefix: "v1beta1-conditions-c",
-				Inject: true,
-			})
-
+			nsA := apps.Namespace1
+			nsB := apps.Namespace2
+			nsC := apps.Namespace3
+			a := apps.A.Match(echo.NameSpace(nsA.Name()))
+			b := apps.B.Match(echo.NameSpace(nsB.Name()))
+			c := apps.C.Match(echo.NameSpace(nsC.Name()))
 			portC := 8090
-			var a, b, c echo.Instance
-			echoboot.NewBuilder(ctx).
-				With(&a, util.EchoConfig("a", nsA, false, nil, nil)).
-				With(&b, util.EchoConfig("b", nsB, false, nil, nil)).
-				With(&c, echo.Config{
-					Service:   "c",
-					Namespace: nsC,
-					Subsets:   []echo.SubsetConfig{{}},
-					Ports: []echo.Port{
-						{
-							Name:         "http",
-							Protocol:     protocol.HTTP,
-							InstancePort: portC,
-						},
-					},
-				}).
-				BuildOrFail(t)
-
 			args := map[string]string{
 				"NamespaceA": nsA.Name(),
 				"NamespaceB": nsB.Name(),
 				"NamespaceC": nsC.Name(),
-				"IpA":        getWorkload(a, t).Address(),
-				"IpB":        getWorkload(b, t).Address(),
-				"IpC":        getWorkload(c, t).Address(),
+				"IpA":        getWorkload(a[0], t).Address(),
```

Comment:
Would it make sense to extend the template with the `range` operator so you can use the `sourceIP` from each cluster? Seems like v1beta1-conditions will only work when you have cluster-0 as the source cluster.